unstructured/partition/code.py

```diff
@@ -0,0 +1,217 @@
+    FileType.GO: "go",
+    FileType.PY: "python",
+    FileType.CPP: "cpp",
+    FileType.JS: "javascript",
+    FileType.TS: "typescript",
+}
+
+def partition_code(
+    filename: Optional[str] = None,
+    file: Optional[IO[bytes]] = None,
+    text: Optional[str] = None,
+    encoding: Optional[str] = None,
+    language: Optional[str] = None,
+    max_partition: Optional[int] = 1500,
+    min_partition: Optional[int] = 0,
+    metadata_last_modified: Optional[str] = None,
+    chunking_strategy: Optional[str] = None,
+    detection_origin: Optional[str] = "text",
+    **kwargs: Any,
+) -> List[Element]:
+    """Partitions an .txt documents into its constituent paragraph elements.
+    If paragraphs are below "min_partition" or above "max_partition" boundaries,
+    they are combined or split.
+    Parameters
+    ----------
+    filename
+        A string defining the target filename path.
+    file
+        A file-like object using "rb" mode --> open(filename, "rb").
+    text
+        The string representation of the .txt document.
+    encoding
+        The encoding method used to decode the text input. If None, utf-8 will be used.
+    max_partition
+        The maximum number of characters to include in a partition. If None is passed,
+        no maximum is applied.
+    min_partition
+        The minimum number of characters to include in a partition.
+    metadata_last_modified
+        The day of the last modification
+    """
+    return _partition_code(
+        filename=filename,
+        file=file,
+        text=text,
+        encoding=encoding,
+        language=language,
+        max_partition=max_partition,
+        min_partition=min_partition,
+        metadata_last_modified=metadata_last_modified,
+        chunking_strategy=chunking_strategy,
+        detection_origin=detection_origin,
+        **kwargs,
+    )
+
+
+@process_metadata()
+def _partition_code(
+    filename: Optional[str] = None,
+    file: Optional[IO[bytes]] = None,
+    text: Optional[str] = None,
+    metadata_filename: Optional[str] = None,
+    include_metadata: bool = True,
+    languages: Optional[List[str]] = ["auto"],
+    max_partition: Optional[int] = 1500,
+    min_partition: Optional[int] = 50,
+    metadata_last_modified: Optional[str] = None,
+    detect_language_per_element: bool = False,
+    detection_origin: Optional[str] = "codefile",
+    **kwargs: Any,
+) -> List[Element]:
+    """internal API for `partition_code`"""
+    if text is not None and text.strip() == "" and not file and not filename:
+        return []
+
+    if (
+        min_partition is not None
+        and max_partition is not None
+        and (min_partition > max_partition or min_partition < 0 or max_partition < 0)
+    ):
+        raise ValueError("Invalid values for min_partition and/or max_partition.")
+
+    # Verify that only one of the arguments was provided
+    exactly_one(filename=filename, file=file, text=text)
+    file_text = bytes()
+    language = None
```

Comment:
Sure thing,
Also, for now I've put the programming language as the language metadata, but it might be confusing in the rest of the pipeline, any input on that ?